### DISCUSSION TITLE
Maverick ET73 random temperatures

### DISCUSSION THREAD

**Discussion by DaveDixon** [merbanan/rtl_433#3040] [Q&A]
RTL_433 reports both food temperature (temperature_1_F) and ambient temperature (temperature_2_F) for my Maverick ET73. The ambient temperature agrees with the transmitter all the time. The food temperature switches between the temperature on the transmitter and a temperature 20 to 30 degrees higher. Is this a known thing with this device? Thanks!

**Comment by zuckschwerdt** [comment]
Try to use ` rtl_433 -R 114:v` which then will show `[maverick_et73_decode] codes` (and additionally `[pulse_slicer_ppm] Maverick et73 codes` which should match).
Post some broken codes here and we can take a look.

**Comment by DaveDixon** [comment]
Here's the output

```
time      : 2024-08-25 11:23:58
model     : Maverick-ET73 Random Id : 213
Temperature 1: 43.2 C    Temperature 2: 24.9 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:24:00
model     : Maverick-ET73 Random Id : 213
Temperature 1: 43.2 C    Temperature 2: 24.9 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:25:33
model     : Maverick-ET73 Random Id : 213
Temperature 1: 35.2 C    Temperature 2: 24.9 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:29:09
model     : Maverick-ET73 Random Id : 213
Temperature 1: 28.8 C    Temperature 2: 25.1 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:29:11
model     : Maverick-ET73 Random Id : 213
Temperature 1: 28.8 C    Temperature 2: 25.1 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:29:30
model     : Maverick-ET73 Random Id : 70
Temperature 1: 30.4 C    Temperature 2: 25.3 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:29:32
model     : Maverick-ET73 Random Id : 70
Temperature 1: 30.4 C    Temperature 2: 25.3 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:29:34
model     : Maverick-ET73 Random Id : 70
Temperature 1: 30.4 C    Temperature 2: 25.3 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:34:18
model     : Maverick-ET73 Random Id : 70
Temperature 1: 24.0 C    Temperature 2: 25.8 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:34:19
model     : Maverick-ET73 Random Id : 70
Temperature 1: 24.0 C    Temperature 2: 25.8 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:36:46
model     : Maverick-ET73 Random Id : 70
Temperature 1: 33.6 C    Temperature 2: 26.0 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:36:50
model     : Maverick-ET73 Random Id : 70
Temperature 1: 41.6 C    Temperature 2: 26.0 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:36:53
model     : Maverick-ET73 Random Id : 70
Temperature 1: 43.2 C    Temperature 2: 26.1 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:36:56
model     : Maverick-ET73 Random Id : 70
Temperature 1: 36.8 C    Temperature 2: 26.0 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:37:01
model     : Maverick-ET73 Random Id : 70
Temperature 1: 36.8 C    Temperature 2: 26.0 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:37:04
model     : Maverick-ET73 Random Id : 70
Temperature 1: 46.4 C    Temperature 2: 26.0 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:37:08
model     : Maverick-ET73 Random Id : 70
Temperature 1: 49.6 C    Temperature 2: 26.0 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:37:14
model     : Maverick-ET73 Random Id : 70
Temperature 1: 35.2 C    Temperature 2: 25.9 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:37:26
model     : Maverick-ET73 Random Id : 70
Temperature 1: 49.6 C    Temperature 2: 25.9 C
``` 

I restarted the transmitter at 11:29:30 in order to sync with the receiver, hence the change in Random Id. The receiver shows 25C at all times for both probes.

I started holding the Temperature 1 probe in my hand at 11:36:46. The receiver went steadily up to 35C for Temperature 1.

I swapped the probes at 11:40:35. The display showed Temperature 1 = 25, Temperature 2 = 29

```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:40:35
model     : Maverick-ET73 Random Id : 70
Temperature 1: 28.8 C    Temperature 2: 28.5 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:40:48
model     : Maverick-ET73 Random Id : 70
Temperature 1: 28.8 C    Temperature 2: 28.0 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:41:15
model     : Maverick-ET73 Random Id : 70
Temperature 1: 28.8 C    Temperature 2: 27.5 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:41:54
model     : Maverick-ET73 Random Id : 70
Temperature 1: 28.8 C    Temperature 2: 27.0 C
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2024-08-25 11:43:00
model     : Maverick-ET73 Random Id : 70
Temperature 1: 30.4 C    Temperature 2: 26.5 C
```

**Comment by DaveDixon** [comment]
I tried again with `rtl_433 -v -R 114`

Nothing about pulse_slicer_ppm

Incidentally, this is version 20.02-1, the current version for Ubuntu 20.04.6

There is another RTLSDR device decoding 900 MHz signals. The problem with the Maverick occurred even before adding the second RTLSDR device.

```
rtl_433 version unknown inputs file rtl_tcp RTL-SDR SoapySDR
Use -h for usage help and see https://triq.org/ for documentation.
Trying conf file at "rtl_433.conf"...
Trying conf file at "/home/ddixon/.config/rtl_433/rtl_433.conf"...
Trying conf file at "/usr/local/etc/rtl_433/rtl_433.conf"...
Trying conf file at "/etc/rtl_433/rtl_433.conf"...
Registering protocol [114] "Maverick et73"
Registered 1 out of 149 device decoding protocols
Found 2 device(s)

<excluded>

trying device  1:  Realtek, RTL2838UHIDIR, SN: 00000001
Found Rafael Micro R820T tuner
Using device 1: Generic RTL2832U OEM
Exact sample rate is: 250000.000414 Hz
Enabled direct sampling mode, input 2
Sample rate set to 250000 S/s.
Bit detection level set to 0 (Auto).
Tuner gain set to Auto.
Reading samples in async mode...
[R82XX] PLL not locked!
Disabled direct sampling mode
Tuned to 433.920MHz.

```

**Reply by zuckschwerdt** [reply]
That version is too old to report the raw codes with `rtl_433 -R 114:v`
You can record samples with `-S known` then play the back with rtl_433 and look for one good sample and one showning the error, we can work with that.
Otherwise compile a current version to get the raw codes.

**Reply by DaveDixon** [reply]
Will the current version compile on Ubuntu 20.04?

**Reply by zuckschwerdt** [reply]
Yes, but you can also get the compiled pre-release version from https://github.com/merbanan/rtl_433/releases
You likely want  rtl_433-rtlsdr-openssl3-Linux-amd64-nightly.zip

**Reply by DaveDixon** [reply]
Downloaded rtl_433-rtlsdr-openssl3-Linux-amd64-nightly.zip.

rtl_433: error while loading shared libraries: libssl.so.3: cannot open shared object file: No such file or directory

**Reply by DaveDixon** [reply]
Using -S known.

I didn't get any readings that were near to correct until I put the probes in an oven. The two attached are close.cu8 (Temperature 1: 28.8 C, Temperature 2: 28.3 C, Receiver 28C) and far.cu8 (Temperature 1: 100.8 C, Temperature 2: 72.7 C, Receiver 73C)

[rtl433_output.zip](https://github.com/user-attachments/files/16741677/rtl433_output.zip)

**Comment by zuckschwerdt** [comment]
> rtl_433: error while loading shared libraries: libssl.so.3: cannot open shared object file: No such file or directory

The openssl11 version should work then ;)

**Reply by zuckschwerdt** [reply]
If we put both codes you got with those samples in [a BitBench](https://triq.net/bitbench#c=4612211bc0bb%20%5B1%3A%2028.8%20C%2C%202%3A%2028.3%20C%2C%20Receiver%2028C%5D&c=4632d2d7c0f1%20%5B1%3A%20100.8%20C%2C%202%3A%2072.7%20C%2C%20Receiver%2073C%5D&f=ID%3A8h%20TEMP_A%3A12d%20TEMP_B%3A12d%20CHK%3F16h%0AID%3A8h%20TEMP_A%3A12b%20TEMP_B%3A12b%20CHK%3F16h%0A&z=1&cw=4) we don't see anything weird like an obvious overflow flag or something.

The next to last byte might be flags to indicate the quality of readings or such, we have not found a meaning. Also the last byte should be some checksum but we have not found the mapping. If you have consecutive readings with very little changes (i.e. only 0.1 C changes) we might be able to figure those out.

Not sure why the messages have this strange data then. Do good a and bad readings alternate when staying on the same temperatures? Can you describe when the bad readings might occur?

**Reply by DaveDixon** [reply]
Yesterday I was running the smoker for 5 hours, maintaining smoker temperature (Temperature 2) of 240F. Temperature 1 on mqtt jumped around but averaged about 15F above what the receiver reported, except occasionally it showed the same temperature as the receiver. The short run workaround was to keep track of the lowest temperature reported over the past 20 minutes, say. The receiver, however, was always stable and showed the correct temperature as verified by a quick-read thermometer.

I will run another test in the oven tomorrow.

**Reply by zuckschwerdt** [reply]
**edit: wrong issue, sorry.**

**Reply by zuckschwerdt** [reply]
**edit: wrong issue, sorry.**

**Comment by DaveDixon** [comment]
Due to other problems, I had to stand up a new server. This one runs Ubuntu 24.04 with the default rtl_433 version 23.11. 

I haven't run another oven test, but I am running the smoker today. The temperature readings have been very stable and consistent with the receiver. So it appears this issue has been overtaken by events.

**Reply by zuckschwerdt** [reply]
Happy to hear. If you encounter any other weirdness just save some raw codes to analyze and maybe refine the decoder.
